clientapi/routing/sendevent.go

```diff
@@ -120,6 +127,46 @@ func SendEvent(
 	return res
 }
 
+func checkPowerLevels(
+	ctx context.Context,
+	roomID string,
+	rsAPI api.RoomserverInternalAPI,
+	event *gomatrixserverlib.Event,
+) (ok bool, response *util.JSONResponse) {
+	req := api.QueryLatestEventsAndStateRequest{RoomID: roomID}
+	res := api.QueryLatestEventsAndStateResponse{}
+	if err := rsAPI.QueryLatestEventsAndState(ctx, &req, &res); err != nil {
+		return false, &util.JSONResponse{
+			Code: http.StatusInternalServerError,
+			JSON: err.Error(),
+		}
+	}
+
+	var creator string
+	for _, v := range res.StateEvents {
+		pl, _ := v.PowerLevels() // we only care about the existence of power levels, so ignore any errors
+		if pl != nil {
+			return true, nil
+		}
+		if v.Type() == gomatrixserverlib.MRoomCreate {
+			creator = v.Sender()
+		}
+	}
+
+	if creator != "" && event.Sender() != creator {
```

Comment:
I don't know if we need a check in the client API for this? The roomserver should reject an event that it isn't happy with, and the client API won't be involved in remote servers pushing state to us.